### ISSUE TITLE
VS Code for Eclipse Che works with both UBI 8 and 9

### ISSUE THREAD

**Issue by azatsarynnyy** [eclipse/che#21629]
### Is your task related to a problem? Please describe

When running Che-Code in a UBI9-based container, it fails with the following error:
```
./node: error while loading shared libraries: libcrypto.so.1.1: cannot open shared object file: No such file or directory 
```

### Describe the solution you'd like

Probably, it requires adding some libs to Che-Code image.

### Describe alternatives you've considered

There was an internal discussion on the possibility of running Che-Code in any image. It requires reviewing how we package the VS Code. Also, there were some ideas to check about using the statically linked libraries.

### Release Notes Text

Previously VS Code for Eclipse Che was failing to run on containers that had a version of OpenSSL that was not supported like for example UBI 9 that included OpenSSL 3.0.1). Now it will work independently from the version of OpenSSL included in the container.

**Comment by benoitf** [comment]
on ubi9 there are different libcrypto versions
```
/usr/lib64/libcrypto.so.3.0.1
/usr/lib64/libcrypto.so.3
```

**Comment by benoitf** [comment]
so maybe copying nodejs binary from `ubi9/nodejs-16` in addition to `ubi8` or we grab a nodejs binary with a different openssl version

**Comment by nickboldt** [comment]
In order for this to work downstream we would need to start from https://catalog.redhat.com/software/containers/ubi9/nodejs-16/61a60604c17162a20c1c6a2e or https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5 and adjust the image as needed.

**Comment by l0rd** [comment]
> In order for this to work downstream we would need to start from [catalog.redhat.com/software/containers/ubi9/nodejs-16/61a60604c17162a20c1c6a2e](https://catalog.redhat.com/software/containers/ubi9/nodejs-16/61a60604c17162a20c1c6a2e) or [catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5](https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5) and adjust the image as needed.

@nickboldt are you sure? This is not about using UBI 9 image for UDI (or for VS Code). This is to allow a user to specify an UBI 9 based custom image in a devfile and VS Code as an editor.

**Comment by RomanNikitenko** [comment]
`UBI 8` image uses `1.1.1` version of `openssl` 

<img width="746" alt="image" src="https://user-images.githubusercontent.com/5676062/191010411-ad9f110a-a18b-42e7-b333-351c59cab1e3.png">

`UBI 9` image uses `3.0.1` version of `openssl`

<img width="746" alt="image" src="https://user-images.githubusercontent.com/5676062/191010144-caf8eaf1-cdb3-471d-8ba2-f2e2bb848b44.png">

For `che-code` we use `node` that requires `1.1.1` version of `openssl`

![image](https://user-images.githubusercontent.com/5676062/191010982-d0e194a9-9670-4ab5-81a0-a9e3035cbb1f.png)

**Comment by azatsarynnyy** [comment]
@l0rd @benoitf we're going to apply the solution proposed in https://github.com/che-incubator/che-code/pull/120. See more details [in the comment](https://issues.redhat.com/browse/CRW-3261?focusedCommentId=21012362&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-21012362). It's simple and allows to work around the original problem while we're not migrated to UBI9.

**Comment by benoitf** [comment]
I would have added a nodejs ubi9 binary in the image and pick-up at runtime the correct nodejs binary.

if ssl 1.1, pick-up nodejs from ubi8 else if ssl 3, use nodejs from ubi9 (we already have a nodejs for alpine and another one for ubi8)

Also the active nodejs LTS version will be switched to nodejs 18 at the end of this month (and since nodejs17, openssl to use is v3) so more and more people will expect to have openssl 3

**Comment by RomanNikitenko** [comment]
> I would have added a nodejs ubi9 binary in the image and pick-up at runtime the correct nodejs binary.

hello, Florent!

Agree, it would be good solution and I tried it some time ago. 
I took `Node.js 16` which statically links against `OpenSSL 3` from  `ubi9/nodejs-16` image and tried to use it [here](https://github.com/che-incubator/che-code/pull/120/files#diff-f01fcd4f0c7a628894e56bb331ac2b4e43d85e48c51ef93d4ba60ca1ed56a5eeR82) to run VS Code.
But I faced with the error like: `Error: libnode.so.93: cannot open shared object file: No such file or directory`.

I guess the problem is:
- I took installed node file
- but there is no `/usr/lib64/libnode.so.93` file in the customer's runtime image

> Also the active nodejs LTS version will be switched to nodejs 18 at the end of this month (and since nodejs17, openssl to use is v3) so more and more people will expect to have openssl 3

I commented about it 5 days ago in another issue (Artem posted [the corresponding link](https://issues.redhat.com/browse/CRW-3261?focusedCommentId=21012362&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-21012362) above, but you should be logged in).
In short, right now we can not switch from `Node.js 16` to another version as VS Code requires a version >=16.14.x and <17.

**Comment by azatsarynnyy** [comment]
Downstream link: https://issues.redhat.com/browse/CRW-3261

**Comment by l0rd** [comment]
@azatsarynnyy can we close this? Asking to decide if we can include it in the 7.55 release notes

**Comment by l0rd** [comment]
@RomanNikitenko @azatsarynnyy I have changed the title to reflect the actual new feature as it will be mentioned in the release notes. Can you please confirm it's correct?

**Comment by azatsarynnyy** [comment]
> VS Code for Eclipse Che is bundled with OpenSSL 3 (and works with both UBI 8 and 9)

@l0rd VS Code for Eclipse Che is not bundled with OpenSSL 3 after fixing this issue. But OpenSSL 3 is already provided in UBI 9 image.
@RomanNikitenko please, correct me if I'm wrong.

I'd restate the title as `VS Code for Eclipse Che works with both UBI 8 and 9`.

**Comment by max-cx** [comment]
Sync'd with Red Hat JIRA https://issues.redhat.com/browse/CRW-3633

**Comment by ArvinB** [comment]
@SDawley Are there any plans for 2024 to move Dev Spaces from UBI8 to UBI9 base images?
FYI @nickboldt
